liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/GenerateFeaturesMojo.java

```diff
@@ -461,7 +461,7 @@ public String getEEVersion(List<MavenProject> mavenProjects) {
                     if (ver != null) {
                         eeVersionsDetected.add(ver);
                     } else {
-                        // EE version out of range, add max possible verison
+                        // EE version out of range, add max possible version
                         eeVersionsDetected.add(BINARY_SCANNER_UMBRELLA_DEP_MAXV);
```

Comment:
It looks like `getMPVersion` can still encounter a `null` value here - https://github.com/OpenLiberty/ci.maven/blob/8b3da42bd86195596d2d7dddabba840b951d9691/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/GenerateFeaturesMojo.java#L540 - because `composeMPVersion` can return a `null` value. 

If so, is that behaviour correct? Would we want to use the max level MP version if we got a `null` return from `composeMPVersion`? I suppose that may make sense if no other MP versions have been found up to that point.

Is the reason we return `null` from `composeMPVersion` because the `split` method may not work as expected?

Reply:
`composeMPVersion` can return `null` now because the version number is mal-formed. It shouldn't be possible but if the version number was a single digit split would fail. So perhaps the correct behaviour now is to ignore null.

Reply:
We will open another issue to design a better error handling mechanism.